Bisect script failures

For any Scala source, let us call it `Test.scala` the following fails 
```
$ scala-cli project/scripts/bisect.scala -- compile Test.scala
```

```
Bisecting with validation script: /var/folders/j_/1jnflqdn6vg_6q2602rxq46m0000gn/T/scala-bisect-validator9193156609557343888
#####################################

#!/usr/bin/env bash
scala-cli compile -S "$1" --server=false Test.scala


#####################################

Verifying the first release: 3.0.1-RC1-bin-20210413-f3c1468-NIGHTLY
Testing 3.0.1-RC1-bin-20210413-f3c1468-NIGHTLY
Checking https://repo1.maven.org/maven2/org/scala-lang/scala3-library_3/
Checked https://repo1.maven.org/maven2/org/scala-lang/scala3-library_3/
Checking https://repo1.maven.org/maven2/org/scala-lang/scala3-library_3/maven-metadata.xml
Checked https://repo1.maven.org/maven2/org/scala-lang/scala3-library_3/maven-metadata.xml
19 is not a valid choice for -release
  scalac -help  gives more information
Compilation failed
Test result: 3.0.1-RC1-bin-20210413-f3c1468-NIGHTLY is a bad release

Exception in thread "main" java.lang.AssertionError: assertion failed: The evaluation script unexpectedly failed for the first checked release
        at scala.runtime.Scala3RunTime$.assertFailed(Scala3RunTime.scala:8)
        at ReleaseBisect.verifyEdgeReleases(bisect.scala:203)
        at bisect$package$.run(bisect.scala:64)
        at run.main(bisect.scala:53)
```

--- a/project/scripts/bisect.scala
+++ b/project/scripts/bisect.scala
@@ -1,3 +1,4 @@
+//> using jvm 17 // Maximal JDK version which can be used with all Scala 3 versions, can be overriden via command line arguments '--jvm=21'
 /*
 This script will bisect a problem with the compiler based on success/failure of the validation script passed as an argument.
 It starts with a fast bisection on released nightly builds.

--- a/project/scripts/bisect.scala
+++ b/project/scripts/bisect.scala
@@ -124,6 +124,7 @@ object ValidationScript:
 
   def tmpScalaCliScript(command: String, args: Seq[String]): File = tmpScript(s"""
     |#!/usr/bin/env bash
+    |export JAVA_HOME=${sys.props("java.home")}
     |scala-cli ${command} -S "$$1" --server=false ${args.mkString(" ")}
     |""".stripMargin
   )
@@ -242,6 +243,7 @@ class CommitBisect(validationScript: File, shouldFail: Boolean, bootstrapped: Bo
     val bisectRunScript = raw"""
       |scalaVersion=$$(sbt "print ${scala3CompilerProject}/version" | tail -n1)
       |rm -rf out
+      |export JAVA_HOME=${sys.props("java.home")}
       |sbt "clean; set every doc := new File(\"unused\"); set scaladoc/Compile/resourceGenerators := (\`${scala3Project}\`/Compile/resourceGenerators).value; ${scala3Project}/publishLocal"
       |${validationCommandStatusModifier}${validationScript.getAbsolutePath} "$$scalaVersion"
     """.stripMargin

--- a/project/scripts/bisect.scala
+++ b/project/scripts/bisect.scala
@@ -244,8 +244,9 @@ class CommitBisect(validationScript: File, shouldFail: Boolean, bootstrapped: Bo
       |scalaVersion=$$(sbt "print ${scala3CompilerProject}/version" | tail -n1)
       |rm -rf out
       |export JAVA_HOME=${sys.props("java.home")}
-      |sbt "clean; set every doc := new File(\"unused\"); set scaladoc/Compile/resourceGenerators := (\`${scala3Project}\`/Compile/resourceGenerators).value; ${scala3Project}/publishLocal"
-      |${validationCommandStatusModifier}${validationScript.getAbsolutePath} "$$scalaVersion"
+      |(sbt "clean; set every doc := new File(\"unused\"); set scaladoc/Compile/resourceGenerators := (\`${scala3Project}\`/Compile/resourceGenerators).value; ${scala3Project}/publishLocal" \
+      |  || (echo "Failed to build compiler, skip $$scalaVersion"; git bisect skip) \
+      |) && ${validationCommandStatusModifier}${validationScript.getAbsolutePath} "$$scalaVersion"
     """.stripMargin
     "git bisect start".!
     s"git bisect bad $fistBadHash".!

